### PR TITLE
1 user dashboard swb

### PR DESCRIPTION
**Pull request by Scott-Borecki**
## Changes Implemented:
  - Completed: 
    - #3 
    - #19 
    - #20 
  - Added missing test for `User` poro
  - Updated attributes for `UserEvent` poro to match API response and added missing test
  - Added missing test for `YelpGym` poro
  - Updated attributes for `UserGym` poro (need to add missing test)
  - I removed the partial for the profile.  We can discuss these as a team.  My reasoning behind that is that we can refactor and abstract to a partial after we have everything functioning and can look for redundencies.  I felt like the partial was adding too much complexity to this user story / dashboard development and was painting ourselves into a corner.

## Quality Control Checklist:
  - [ ] Code adheres to Rubocop styling (if unavoidable infractions, please clarify below)
    - There are 17 offenses; some of which are related to using an instance variable for `current_user`
  - [ ] 100% SimpleCov test coverage (if below, please clarify below)
    - 132/134 (98.51%)
    - `BackEndService.create_user` not being hit
    - `ApplicationController.current_user` not being hit
  - [X] Last commit has passing Travis build

## Next Steps & Additional Notes:
- Make links on dashboard functional
- Check if there are any merge conflicts with other PRs and address each
- There may be some room to refactor and combine the redundent `YelpGym` and `UserGym` poros.  Perhaps we can combine this into just a `Gym` poro and only populate the attributes we need for each use case.

### CODE REVIEW

File: app/facades/back_end_facade.rb [hunk 1]
```diff
@@ -11,10 +11,10 @@ def create_user(params)
...
     def get_user_friends(user_id)
       friends = BackEndService.get_friends(user_id)
-      if friends[:data]
-        friends[:data].map do |friend|
-          User.new(friend[:attributes])
-        end
+      return unless friends[:data]
+
+      friends[:data].map do |friend|
+        User.new(friend[:attributes])
```
Comment (tvaroglu) [inline app/facades/back_end_facade.rb:17]: Swag!

File: spec/features/users/dashboard_spec.rb [hunk 2]
```diff
@@ -1,32 +1,240 @@
...
+        gym_membership_id: 3,
+        date_time: '2022-09-22T21:41:28.289Z',
+        activity: 'Stretching'
+      }
+    }
+  end
+
+  let(:user_friends) { [User.new(user1_params), User.new(user2_params), User.new(user3_params)] }
+  let(:user_gyms) { [YelpGym.new(gym1_params), YelpGym.new(gym2_params), YelpGym.new(gym3_params)] }
+  let(:user_events) { [UserEvent.new(event1_params), UserEvent.new(event2_params), UserEvent.new(event3_params)] }
```
Comment (tvaroglu) [inline spec/features/users/dashboard_spec.rb:131]: NICE test setup, love this

**Review by ealwafai — APPROVED** [review]
Looks great! i'm not too familiar with using let in our test though

**Comment by tvaroglu** [comment]
> Looks great! i'm not too familiar with using let in our test though

`let` us show you the way ;)

File: spec/features/users/dashboard_spec.rb [hunk 3]
```diff
@@ -1,32 +1,240 @@
...
+      it 'displays the users I follow', :vcr do
+        expect(page).to have_css('#friends')
+
+        within '#friends' do
+          expect(page).to have_content('My Friends')
+
+          user_friends.each do |friend|
+            expect(page).to have_css("#friend-#{friend.id}")
+
+            within "#friend-#{friend.id}" do
```
Comment (carolinectan) [inline spec/features/users/dashboard_spec.rb:169]: Nice use of within blocks

**Review by carolinectan — APPROVED** [review]

File: app/facades/back_end_facade.rb [hunk 4]
```diff
@@ -2,7 +2,9 @@ class BackEndFacade
   class << self
     def get_user(google_id)
       user = BackEndService.get_user(google_id)
-      user[:data].present? ? User.new(user[:data]) : nil
+      return unless user[:data]
+
+      User.new(user[:data])
```
Comment (bfl3tch) [inline app/facades/back_end_facade.rb:7]: Is there a performance advantage over a ternary?

File: spec/features/users/dashboard_spec.rb (hunk 3 above)
Comment (bfl3tch) [reply spec/features/users/dashboard_spec.rb:169]: Agreed on the within blocks

**Review by bfl3tch — APPROVED** [review]
Looks great!

**Pull request merged**